src/main/java/seedu/address/logic/commands/transaction/DeleteTransactionCommand.java

```diff
@@ -0,0 +1,72 @@
+package seedu.address.logic.commands.transaction;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.UniCashMessages;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.transaction.Transaction;
+
+/**
+ * This class handles the deletion of a Transaction from the UniCa$h database.
+ */
+public class DeleteTransactionCommand extends Command {
+
+    public static final String COMMAND_WORD = "delete_transaction";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the transaction identified by the index number used in the displayed transaction list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_DELETE_TRANSACTION_SUCCESS = "Deleted Transaction: %1$s";
+
+    private final Index targetIndex;
+
+    public DeleteTransactionCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Transaction> lastShownList = model.getFilteredTransactionList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(UniCashMessages.MESSAGE_INVALID_TRANSACTION_DISPLAYED_INDEX);
+        }
+
+        Transaction transactionToDelete = lastShownList.get(targetIndex.getZeroBased());
+        model.deleteTransaction(transactionToDelete);
+        return new CommandResult(String.format(MESSAGE_DELETE_TRANSACTION_SUCCESS,
+                UniCashMessages.formatTransaction(transactionToDelete)));
```

Comment:
Can try adding a unit test for this section to test for the success path? 

```
        Transaction transactionToDelete = model.getFilteredTransactionList().get(INDEX_FIRST_TRANSACTION.getZeroBased());
        DeleteTransactionCommand deleteCommand = new DeleteTransactionCommand(INDEX_FIRST_PERSON);

        String expectedMessage = String.format(DeleteTransactionCommand.MESSAGE_DELETE_PERSON_SUCCESS,
                Messages.format(transactionToDelete));

        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs(), new UniCash());
        expectedModel.deleteTransaction(transactionToDelete);

        assertCommandSuccess(deleteTransactionCommand, model, expectedMessage, expectedModel);
```

Reply:
Done